### PR TITLE
external/mbedtls : enable debugging function call in mbedtls

### PR DESCRIPTION
**Pull request by an4967**
- Remove '#if' statement for general debugging function

### CODE REVIEW

File: external/grpc/src/core/tsi/ssl_transport_security.cc [hunk 1]
```diff
@@ -154,7 +154,7 @@ typedef struct {
 /* Debug for mbedtls SSL process */
 static void grpc_tsi_debug(void *ctx, int level, const char *file, int line, const char *str)
 {
-  (void) ctx;
+  printf("%s(%d) : %s", file, line, str);
```
Comment (pillip8282) [inline external/grpc/src/core/tsi/ssl_transport_security.cc]: Calling the grpc debug method would be better.

**Comment by an4967** [comment]
@pillip8282 I sperated the PR in mbedtls part and grpc part.
So, I want mbedtls fix would be reviewed first.

File: external/mbedtls/debug.c [hunk 2]
```diff
@@ -81,17 +81,13 @@ static inline void debug_send_line(const mbedtls_ssl_context *ssl, int level, co
 	 * Since there is no portable way to get one, use the address of the ssl
 	 * context instead, as it shouldn't be shared between threads.
 	 */
-#if 0							/* For TinyAra */
 #if defined(MBEDTLS_THREADING_C)
```
Comment (pradeepns85) [inline external/mbedtls/debug.c]: Is it really required to make this API as inline function ? it's been called from several functions and it just increases code space.  May not be directly linked to your changes but am just mentioning the overhead of removing #if 0

Comment (an4967) [reply external/mbedtls/debug.c]: @pradeepns85 
if you see line 103 in debug.c, it always checks 'ssl->conf->f_dbg' to be registered by users.
As following this logic, even if an user registered f_dbg, it doesn't work and wastes code space. 
Moreover, if 'f_dbg' is not set, printing debugging logs fails.

**Comment by seinfra** [comment]
Target : [ebcf4359e824045a0b1913ea245ea051ae3d3fc4] - Code Rule Check (C++) OK.

**Pull request merged**